subsys/bluetooth/shell/tbs_client.c

```diff
@@ -51,27 +53,44 @@ static int cmd_tbs_client_set_signal_strength_interval(const struct shell *sh,
 						       size_t argc,
 						       char *argv[])
 {
-	int result;
-	int interval;
-	int inst_index;
+	unsigned long inst_index;
+	unsigned long interval;
+	int result = 0;
 
 	if (argc > 2) {
 		if (strcmp(argv[1], "gtbs") == 0) {
 			inst_index = BT_TBS_GTBS_INDEX;
 		} else {
-			inst_index = strtol(argv[1], NULL, 0);
-			if (inst_index < 0 || inst_index > UINT8_MAX) {
-				shell_error(sh, "Invalid index");
+			inst_index = shell_strtoul(argv[1], 0, &result);
```

Comment:
add a comment/TODO that this can be pulled out as a function since this code is repeated many times?